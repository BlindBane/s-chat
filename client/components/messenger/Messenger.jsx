import React, { Component } from 'react'
import MessageBox from './MessageBox'
import MessageSearchBar from './MessageSearchBar'

class Messenger extends Component {
  state = {
    searchTerm: '',
    messages: this.props.messages
  }
  handleSearchTermChange = (e) => {
    this.setState({ searchTerm: e.target.value })
  }
  handleNewMsg = (newMsg) => {
    const newMsgs = [...this.state.messages.slice(0), newMsg]
    this.setState({
      messages: newMsgs
    })
  }
  render () {
    return (
      <div>
        <MessageSearchBar
          handleSearchTermChange={this.handleSearchTermChange}
          searchTerm={this.state.searchTerm}
        />
        <MessageBox
          messages={this.state.messages}
          handleNewMsg={this.handleNewMsg}
          searchTerm={this.state.searchTerm}
        />
      </div>
    )
  }
}

export default Messenger
