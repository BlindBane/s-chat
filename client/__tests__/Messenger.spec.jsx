import React from 'react'
import { shallow } from 'enzyme'
import Messenger from './../components/messenger/Messenger'

test('Messenger renders correctly', () => {
  const component = shallow(<Messenger />)
  expect(component).toMatchSnapshot()
})
