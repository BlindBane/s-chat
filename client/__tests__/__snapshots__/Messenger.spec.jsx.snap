// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messenger renders correctly 1`] = `
<div>
  <MessageSearchBar
    handleSearchTermChange={[Function]}
    searchTerm=""
  />
  <MessageBox
    handleNewMsg={[Function]}
    searchTerm=""
  />
</div>
`;
